### PR TITLE
TRUNK-5334: Rename allergy.comment to allergy.comments

### PR DESCRIPTION
**Pull request by MadhavKauntia**
## Description of what I changed

I renamed allergy.comment to allergy.comments.

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5334

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by dkayiwa** [comment]
Can you add @since annotations to the new methods?

File: api/src/main/resources/liquibase-update-to-2.0.xml [hunk 1]
```diff
@@ -8810,7 +8810,7 @@
     </changeSet>
     
     <changeSet id="TRUNK-5334-rename-allergy-comment-to-comments" author="madhavkauntia">
-    	<renameColumn tableName="allergy" oldColumnName="comment" newColumnName="comments"/>
+    	<renameColumn tableName="allergy" oldColumnName="comment" newColumnName="comments" columnDataType="varchar(1024)"/>
```
Comment (dkayiwa) [inline api/src/main/resources/liquibase-update-to-2.0.xml]: Is there any reason why you are changing the data type?

Comment (MadhavKauntia) [reply api/src/main/resources/liquibase-update-to-2.0.xml]: I haven't changed the datatype. It's the same. When I tried building and running without specifying the columnDataType attribute, the web page gave an error specifying that that attribute needs to be provided in the <renameColumn> tag. The problem was fixed when I made the required change.

**Comment by MadhavKauntia** [comment]
> Can you add @SInCE annotations to the new methods?

Which version should I specify in the @since annotation?

**Comment by dkayiwa** [comment]
The first version to be released with these changes.

**Comment by MadhavKauntia** [comment]
@since annotations added.

File: api/src/main/java/org/openmrs/Allergy.java [hunk 2]
```diff
@@ -158,16 +158,36 @@ public void setSeverity(Concept severity) {
...
 	 */
+	@Deprecated
 	public void setComment(String comment) {
-		this.comment = comment;
+		setComments(comment);
+	}
+	
+	/**
+	 * @return Returns the comments
+	 * @since 2.1.5
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/Allergy.java]: The version would be 2.3.0

Comment (MadhavKauntia) [reply api/src/main/java/org/openmrs/Allergy.java]: Apologies. The changes have been made.

File: api/src/main/java/org/openmrs/Allergy.java [hunk 3]
```diff
@@ -158,16 +158,36 @@ public void setSeverity(Concept severity) {
 	
 	/**
 	 * @return Returns the comment
+	 * @deprecated use getComments() instead
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/Allergy.java]: Did you take a look at this? https://wiki.openmrs.org/display/docs/Java+Conventions#JavaConventions-Deprecation

File: api/src/main/java/org/openmrs/Allergy.java [hunk 4]
```diff
@@ -158,16 +158,36 @@ public void setSeverity(Concept severity) {
 	
 	/**
 	 * @return Returns the comment
+	 * @deprecated as of 2.2, replaced by {@link #getComments()}
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/Allergy.java]: The deprecation version would be 2.3.0

File: api/src/main/resources/liquibase-update-to-2.3.xml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    v. 2.0. If a copy of the MPL was not distributed with this file, You can
+    obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+    the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+
+    Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+    graphic logo is a trademark of OpenMRS Inc.
+
+-->
+<!--
+    This file contains all changesets that have been introduced on top of liquibase-update-to-2.0.xml exclusively.
```
Comment (dkayiwa) [inline api/src/main/resources/liquibase-update-to-2.3.xml]: Is this comment correct?

**Pull request merged**